clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -293,6 +293,11 @@ public void handshake() throws IOException {
                 updateBytesBuffered(true);
         } catch (SSLException e) {
             maybeProcessHandshakeFailure(e, true, null);
+        } catch (EOFException e) {
+            // it is impossible to complete handshake successfully due to end-of-stream. Hence, it should throw either
+            // EOFException or SslAuthenticationException directly.
+            maybeThrowSslAuthenticationException();
+            throw e;
```

Comment:
@chia7712 Is there a reason why we need this change? If so, can we add a test? If there is no data to unwrap, we would just throw the exception in the IOException catch block below anyway. Since we only get here when the other side has closed its connection, which is typically a handshake failure, I am not sure we need to optimize.

Reply:
@rajinisivaram thanks for the feedback. This optimization is trivial and there is no strong reason for it. So I'm going to close this PR :)